x-pack/plugins/security_solution/server/endpoint/alerts/handlers/list/index.ts

```diff
@@ -18,14 +19,11 @@ export const alertListHandlerWrapper = function (
     res
   ) => {
     try {
-      const indexPattern = await endpointAppContext.service
-        .getIndexPatternRetriever()
-        .getEventIndexPattern(ctx);
```

Comment:
We may want to keep some sort of wrapper such that when the implementation changes in the future we do not have to undergo this change again. This is close to where we started and we are almost back there. I have also seen index definitions in the advanced section of the endpoint data sources. And I am not sure how this will influence this implementation. I think it is fair to abstract this index retrieval away with a thin wrapper if possible.

Reply:
Franklin and I talked offline but for now it'd probably be best to keep things as simple as possible since the index patterns are just strings. If we need to abstract that away in the future, the plumbing with the singleton service is still there and it shouldn't be too difficult to add it back.